gateway-messages/src/sp_to_mgs.rs

```diff
@@ -1075,6 +1078,11 @@ pub enum SpError {
     Watchdog(WatchdogError),
     Monorail(MonorailError),
     Dump(DumpError),
+    // These are common enough they get a specific error
+    HfNotMuxedToSp,
+    HfBadAddress,
+    HfQspiTimeout,
+    HfQspiTransferError,
```

Comment:
it looks like most other operations in the MGS API use nested enums, why not do a ` Hf(HfError)` here instead?